test/middleware/spec.check-progress.js

```diff
@@ -50,4 +51,164 @@ describe('middleware/check-session', function () {
+                    ],
+                    forks: [{
+                        target: '/step4'
+                    }]
+                },
+                '/step3': {
+                    next: '/step4',
+                    fields: [
+                        'step-3-field-1',
+                        'step-3-field-2'
+                    ]
+                },
+                '/fork1': {
+                    next: '/step2',
+                    fields: [
+                        'fork-field-1',
+                        'fork-field-2'
+                    ]
+                },
+                '/step4': {
+                    fields: [
+                        'step-4-field-1',
+                        'step-4-field-2'
+                    ]
+                }
+            };
+
+            sessionFields = {
+                'step-1-field-1': true,
+                'step-1-field-2': true,
+                'step-2-field-1': true,
+                'step-2-field-2': true,
+                'step-3-field-1': true,
+                'step-3-field-2': true,
+                'step-4-field-1': true,
+                'step-4-field-2': true,
+                'fork-field-1': true,
+                'fork-field-2': true
+            };
+
+            sessionSteps = ['/step1', '/step2', '/step3', '/fork1', '/step4'];
+
+            req.sessionModel.set(sessionFields);
+            req.sessionModel.set({
+                steps: sessionSteps
+            });
+
+            controller.options.next = steps['/step1'].next;
+            controller.options.forks = steps['/step1'].forks;
+
+            checkProgress('/step1', controller, steps, '/');
+        });
+
+        describe('when all steps are still reachable after forking', function () {
+            beforeEach(function () {
+                controller.getNextStep = sinon.stub().returns('/fork1');
+                controller.emit('complete', req, res);
+            });
+
+            it('no steps are removed', function () {
+                req.sessionModel.get('steps').should.be.eql(sessionSteps)
+            });
+
+            it('doesn\'t invalidate any fields', function () {
+                req.sessionModel.toJSON().should.contain.all.keys(
+                    'step-1-field-1',
+                    'step-1-field-2',
+                    'step-2-field-1',
+                    'step-2-field-2',
+                    'step-3-field-1',
+                    'step-3-field-2',
+                    'step-4-field-1',
+                    'step-4-field-2',
+                    'fork-field-1',
+                    'fork-field-2'
+                );
+            });
+        });
+
+        describe('When result of forking skips a single step', function () {
+
+            beforeEach(function () {
+              controller.getNextStep = sinon.stub().returns('/step2');
+              controller.emit('complete', req, res);
+            });
+
+            it('invalidates fork1', function () {
```

Comment:
there are 2 forked journeys - step4 and fork one. in this case only fork1 should be invalidated as step4 is still reachable